app/controllers/api/v1/users_controller.rb

```diff
@@ -29,6 +29,48 @@ def update
         end
       end
 
+      # For social media logins, renders the appropriate `redirect_to` path depending on whether the user is registered.
+      #
+      # Requires that the ActionController::Parameters contain a :user key, with a nested :email key.
+      # For example: { user: { email: "john@example.com" } }
+      #
+      # @return [String] A string of the user's redirect_to path
+      # @see https://github.com/zquestz/omniauth-google-oauth2#devise
+      #
+      def exist
+        user = User.find_by(email: params[:user][:email])
+
+        @redirect_path = '/profile'
+        unless user
+          @redirect_path = '/social_login'
+        end
+        render json: { redirect_to: @redirect_path }
+      end
+
+      # For social media logins, creates the user in the database if necessary,
+      # then logs them in, and renders the appropriate `redirect_to` path depending on whether the user is logging in
+      # for the first time.
+      #
+      # @return [String] A string of the user's redirect_to path
+      # @return [Json] A serialied JSON object derived from current_user
+      # @return [Token] A token that the frontend stores to know the user is logged in
+      # @see https://github.com/OperationCode/operationcode_backend/blob/master/app/controllers/api/v1/sessions_controller.rb#L8-L20
+      #
+      def social
+          set_social_user_and_redirect
```

Comment:
Instead of setting instance variables here for `@user` and `@redirect_path` through the `set_social_user_and_redirect` method, which could potentially leak elsewhere in the class, you can have this `User.from_social(params[:user])` method set the two local variable directly here:

```ruby
user, redirect_path = User.from_social(params[:user])
```

Reply:
done.